### PR TITLE
Allow changing of Tabs for Assessments

### PR DESCRIPTION
**Pull request by weiqingtoh**
This implements #2073, allowing individual assessments to be moved from one tab to another. 

I implemented this by making a separate call to GET Categories. @allenwq suggested that we can do the API call only when user attempts to change the tab, but it's a little complex and can be done later, so I left it as a TODO in the code. 

change-tab

### CODE REVIEW

File: client/app/api/course/index.js [hunk 1]
```diff
@@ -12,6 +13,7 @@ import ScribingQuestionAPI from './Assessment/question/scribing';
 const CourseAPI = {
   achievements: new AchievementsAPI(),
   assessment: AssessmentAPI,
+  categories: new CategoriesAPI(),
```
Comment (allenwq) [inline client/app/api/course/index.js]: would be better to nest this under assessment API, follow the controller structure.

File: app/controllers/course/assessment/assessments_controller.rb [hunk 2]
```diff
@@ -70,7 +70,7 @@ def question_order_params
   end
 
   def assessment_params
-    base_params = [:title, :description, :base_exp, :time_bonus_exp, :start_at, :end_at,
+    base_params = [:title, :description, :base_exp, :time_bonus_exp, :start_at, :end_at, :tab_id,
```
Comment (allenwq) [inline app/controllers/course/assessment/assessments_controller.rb]: Do we need to check whether the tab is in the same course with the assessment?

Comment (weiqingtoh) [inline app/controllers/course/assessment/assessments_controller.rb]: How about adding a validation check at the 'Assessment' model instead?

Comment (allenwq) [inline app/controllers/course/assessment/assessments_controller.rb]: 👍

Comment (weiqingtoh) [inline app/controllers/course/assessment/assessments_controller.rb]: Added!

**Pull request merged**